smartsim/_core/_cli/build.py

```diff
@@ -496,3 +499,17 @@ def configure_parser(parser: argparse.ArgumentParser) -> None:
         default=False,
         help="Build KeyDB instead of Redis",
     )
+
+    parser.add_argument(
+        "--torch_with_mkl", 
+        dest="torch_with_mkl", 
+        action="store_true",
+        help="Build Torch with Intel MKL (if available)"
+    )
+    parser.add_argument(
+        "--no_torch_with_mkl",
+        dest="torch_with_mkl",
+        action="store_false",
+        help="Do not build Torch with Intel MKL"
+    )
+    parser.set_defaults(torch_with_mkl=True)
```

Comment:
Not wrong, just be warned: 

I'm not usually a fan of this style of args as this is legal input and will be parsed by `argparse` w/o error
```sh
$ smart build --torch_with_mkl --no_torch_with_mkl
# args.troch_with_mkl == False
$ smart build  --no_torch_with_mkl --torch_with_mkl
# args.torch_with_mkl == True
```
I would vote to return to a single `--no_torch_with_mkl` arg flag with a `'store_false'` directive, similar to how we do `--no_tf`/`--no_pt`/etc., but if we do decide to keep a this style, can we error out if a user were specify both flags just so that they don't have something unexpected happen?